packages/uma-auth-client/src/hooks/nwc-requests/useBalance.tsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from "react";
+import * as Nip47 from "src/Nip47Types";
+import { NwcRequester } from "src/NwcRequester";
+import { useNwcRequester } from "../useNwcRequester";
+
+export const useBalance = () => {
+  const [balance, setBalance] = useState<
+    Nip47.GetBalanceResponse | undefined
+  >();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const { nwcRequester } = useNwcRequester();
+
+  useEffect(() => {
+    const fetchBalance = async (nwcRequester: NwcRequester) => {
+      setIsLoading(true);
+      try {
+        const response = await nwcRequester.getBalance();
```

Comment:
ahh i forgot about this too, should we be using the connection's currency here?

Reply:
Not sure what you mean here; this should be the connection's balance in the same currency as the currency code also provided in the response

Reply:
the getBalance request can take an optional currency code to specify which currency to look up, and the response should be the amount in that currency I believe -- but in this a case I'm not sure if the request currency should be from connection/client app (forgot where we landed on client apps selecting a currency)

Reply:
We landed on client apps not selecting the budget's currency; the connection is always made with the user's default currency (the first currency provided by the vasp)

Reply:
Looking at the nwc backend we pass it along and if it's not provided we just don't pass any params when making the /balance call to the vasp. Did we specify what the vasp should do in that case? cc @yunyuyunyu ?

Reply:
Default with no currency specified is msats since that's the current behavior in nip47

Reply:
Updated to get the balance; refactored useCurrencies in the process to just be `useCurrency` and match the token budget's currencyCode to the get_info currencies data in the same hook